frontend/src/services/canvasDrawTools.tsx

```diff
@@ -99,4 +99,54 @@ export const drawPoints = (context: CanvasRenderingContext2D, canvasSize: number
     context.fillStyle = '#ffffff';
     context.fillText(`${index+1}`, mapCoordinate(point.coordinateX, canvasSize), mapCoordinate(point.coordinateY, canvasSize));
   });
+}
+
+export const drawInterpolation = (context: CanvasRenderingContext2D, canvasSize: number, previousPoints: RoomCalibrationPoint[], colourIndex: number) => {
+  const interpolationDebugResolution = canvasSize; // Or anything where canvasSize % interpolationDebugResolution === 0
+  const powerParam = 1.5;
+
+  const results = new Array(interpolationDebugResolution);
+  let maxResult = 0;
+
+  for (let x = 0; x < interpolationDebugResolution; x++) {
+    for (let y = 0; y < interpolationDebugResolution; y++) {
+      const mappedX = x * maxCord / interpolationDebugResolution;
+      const mappedY = y * maxCord / interpolationDebugResolution;
+      let totalWeight = 0;
+      let totalVolume = 0;
+      let result = 0;
+      for (const roomCalibrationPoint of previousPoints) {
+        if (roomCalibrationPoint.coordinateX === mappedX && roomCalibrationPoint.coordinateY === mappedY) {
+          result = roomCalibrationPoint.measuredVolume;
+          break;
+        }
+
+        const distanceX = roomCalibrationPoint.coordinateX - mappedX;
+        const distanceY = roomCalibrationPoint.coordinateY - mappedY;
```

Comment:
Don't we also have to map the `roomCalibrationPoint` coordinates to the smaller canvas coordinates?

And I was just thinking.. If we map them before the interpolation, we may have a different result here than in the backend. Let's say, one speaker is at x = 400 and we interpolate the border at x = 640, so the distance to the speaker is 240.
Now, in the frontend, we assume that the canvas has 1/2 of the size. So x = 200 and border at 320. The distance is now 120 instead of 240 and when using the same interpolation, it appears louder in the frontend than it actually is in the backend.
Do you understand my concern or am I missing something and this isn't a problem?

Reply:
Ah sorry, I see it now. You are actually covering already both things, I misinterpreted the mapping first 😅